### PR TITLE
fix: navigator.languages to use OS preferred languages

### PR DESCRIPTION
**Pull request by nitsakh**
#### Description of Change
Until recent versions of chrome (i.e. for electron version < 4.x), `navigator.language` returned the system language, and the first item in `navigator.languages` array, was also the same system locale.
However, this behaviour has changed, (it also follows the HTML spec). The language preferences are obtained through the preferences list that chrome maintains. The user has to set these language preferences manually, through the settings menu.
For an electron app, there's no way to set these preferences. Besides, being a desktop app, it seems reasonable for the app to follow the users desktop language preferences.
This change makes the needed changes to set the right OS languages. This also fixes `app.getLocale` which always returned `en-US`.
Adding a spec for this would be difficult, as the os language preferences need to be changed to see the difference.

#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

#### Release Notes

Notes: Fixed behaviour of navigator.language/s and app.getLocale to use OS locale

### CODE REVIEW

File: atom/browser/api/atom_api_web_contents.cc [hunk 1]
```diff
@@ -425,6 +427,23 @@ void WebContents::InitZoomController(content::WebContents* web_contents,
...
+  for (const auto& s16 : languages) {
+    accept_langs += base::SysWideToUTF8(s16) + ',';
+  }
+  accept_langs.pop_back();
+  prefs->accept_languages = accept_langs;
+#else  // OS_LINUX
+  prefs->accept_languages = g_browser_process->GetApplicationLocale();
+#endif
+}
+#endif
```
Comment (nornagon) [inline atom/browser/api/atom_api_web_contents.cc]: Too many `#ifdef`s! Perhaps;

```cpp
void WebContents::SetAcceptLanguages(content::RendererPreferences* prefs) {
#if defined(OS_WIN) || defined(OS_LINUX)
  prefs->accept_languages = GetPreferredOSLanguages();
#else
  NOTREACHED(); // or... what happens on mac?
#endif
}
```
and then define `GetPreferredOSLanguages` in a separate file, possibly using the `foo.h` / `foo_win.cc` / `foo_linux.cc` pattern?

File: atom/browser/atom_browser_main_parts.cc [hunk 2]
```diff
@@ -550,7 +550,7 @@ void AtomBrowserMainParts::PreMainMessageLoopStartCommon() {
 
 #if defined(OS_MACOSX)
   if (custom_locale_.empty())
-    l10n_util::OverrideLocaleWithCocoaLocale();
+    custom_locale_ = GetSystemPreferredLanguage();
```
Comment (nornagon) [inline atom/browser/atom_browser_main_parts.cc]: are you sure we don't still need to call `OverrideLocaleWithCocoaLocale`? Also, do we need to check that the system's preferred language is one that we have localization resources for?

Comment (nitsakh) [reply atom/browser/atom_browser_main_parts.cc]: I think the app should do that, and then choose the next one from the list of preferred languages.

File: atom/browser/api/atom_api_web_contents.cc (hunk 1 above)
Comment (nitsakh) [reply atom/browser/api/atom_api_web_contents.cc]: Yes, I realised that. We can create separate files.

File: atom/app/atom_main_delegate_mac.mm [hunk 3]
```diff
@@ -61,6 +61,11 @@
   base::mac::SetBaseBundleID(base_bundle_id.c_str());
 }
 
+std::string AtomMainDelegate::GetSystemPreferredLanguage() {
```
Comment (jkleinsc) [inline atom/app/atom_main_delegate_mac.mm]: Where does this get used?

Comment (nitsakh) [reply atom/app/atom_main_delegate_mac.mm]: Ah, I think it got lost somewhere in the rebase. I'll update.

**Comment by swapnil0545** [comment]
We are using the Accept-Language Request headers and then setting the app localization accordingly.
We get the required header property in chrome browser but I don't see this in electrons dev tool. I am using electron "3.0.5".

**Comment by zcbenz** [comment]
@nitsakh Do you still want to work on this PR?

**Comment by zcbenz** [comment]
Closing as there has been no work on the PR for a long time, will reopen if anyone wants to continue working on it.

**Pull request closed without merging**